src/main/java/com/github/kagkarlsson/scheduler/SchedulerClient.java

```diff
@@ -25,11 +26,23 @@
 import com.github.kagkarlsson.scheduler.task.TaskInstance;
 
 public interface SchedulerClient {
-	
-	void scheduleForExecution(Instant exeecutionTime, TaskInstance taskInstance);
-	
+
+	/**
+	 * This method has been renamed and will be removed in a future version.
+	 *
+	 * @deprecated use {@link #schedule(Instant, TaskInstance)} instead.
+	 */
+	@Deprecated
+	void scheduleForExecution(Instant executionTime, TaskInstance taskInstance);
+
+	void schedule(Instant executionTime, TaskInstance taskInstance);
+
+	void reschedule(String taskName, String instanceId, Instant newExecutionTime);
```

Comment:
Changes looking good. Starting to feel I lack a type here, a `TaskInstanceId` or something. Previously that was just a `TaskInstance`, but now a `TaskInstance` is something more, it can have data as well. I will think about it, maybe I will change it in a later version :)